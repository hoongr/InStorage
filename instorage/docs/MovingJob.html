<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: MovingJob</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: MovingJob</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>MovingJob<span class="signature">(movingJobID, clientAddress, clientID, hostID, hostAddress, moveDate, moveTime, moverID, inventoryPhotoURL)</span><span class="type-signature"></span></h2>
        
            <div class="class-description">Holds information regarding a moving job between a mover, client, and host</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="MovingJob"><span class="type-signature"></span>new MovingJob<span class="signature">(movingJobID, clientAddress, clientID, hostID, hostAddress, moveDate, moveTime, moverID, inventoryPhotoURL)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Takes the JSON response returned from a Firebase call and
creates a MovingJob from it. Objectifies the JSON.
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>movingJobID</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">this job's unique ID</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>clientAddress</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">client's address</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>clientID</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">client requesting's ID</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>hostID</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">host's ID</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>hostAddress</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">host's address</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>moveDate</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">date that moving job is scheduled for</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>moveTime</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">time that moving job is scheduled for</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>moverID</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">mover's ID</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>inventoryPhotoURL</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">client's inventory</td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_movingJob.js.html">models/movingJob.js</a>, <a href="models_movingJob.js.html#line18">line 18</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="clientAddress"><span class="type-signature"></span>clientAddress<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>MovingJob.clientAddress</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">current clientAddress</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_movingJob.js.html">models/movingJob.js</a>, <a href="models_movingJob.js.html#line32">line 32</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="clientID"><span class="type-signature"></span>clientID<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>MovingJob.clientID</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">current clientID</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_movingJob.js.html">models/movingJob.js</a>, <a href="models_movingJob.js.html#line41">line 41</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="hostAddress"><span class="type-signature"></span>hostAddress<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>MovingJob.hostAddress</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">current hostAddress</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_movingJob.js.html">models/movingJob.js</a>, <a href="models_movingJob.js.html#line50">line 50</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="hostID"><span class="type-signature"></span>hostID<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>MovingJob.hostID</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">current hostID</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_movingJob.js.html">models/movingJob.js</a>, <a href="models_movingJob.js.html#line59">line 59</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="moveDate"><span class="type-signature"></span>moveDate<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>MovingJob.moveDate</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">current moveDate</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_movingJob.js.html">models/movingJob.js</a>, <a href="models_movingJob.js.html#line68">line 68</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="moverID"><span class="type-signature"></span>moverID<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>MovingJob.moverID</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">current moverID</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_movingJob.js.html">models/movingJob.js</a>, <a href="models_movingJob.js.html#line86">line 86</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="moveTime"><span class="type-signature"></span>moveTime<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>MovingJob.moveTime</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">current moveTime</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_movingJob.js.html">models/movingJob.js</a>, <a href="models_movingJob.js.html#line77">line 77</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="BookingRequest.html">BookingRequest</a></li><li><a href="EmailManager.html">EmailManager</a></li><li><a href="FirebaseManager.html">FirebaseManager</a></li><li><a href="Listing.html">Listing</a></li><li><a href="MovingJob.html">MovingJob</a></li><li><a href="UserProfile.html">UserProfile</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.3</a> on Sat Nov 30 2019 19:48:32 GMT-0800 (Pacific Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>