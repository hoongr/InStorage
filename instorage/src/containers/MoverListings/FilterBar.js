import React, { useState, useEffect } from 'react';
import {
  Container,
  Row,
  Col,
  Button,
  Dropdown,
} from 'react-bootstrap';
import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';
import Slider, { Range } from 'rc-slider';
import 'rc-slider/assets/index.css';
import TimePicker from 'rc-time-picker';
import 'rc-time-picker/assets/index.css';
import moment from 'moment';

const now = moment().hour(0).minute(0);

const FilterBar = () => {
  const [startDate, setStartDate] = useState(null);
  const [endDate, setEndDate] = useState(null);
  const [focusedInput, setFocusedInput] = useState(null);

  function updateDates(start, end) {
    setStartDate(start);
    setEndDate(end);
  }
  return (
    <Container
      style={{
        height: "auto",
        borderBottom: "1px solid black",
        padding: "15px",
        position: 'fixed',
        backgroundColor: 'white',
        zIndex: 5,
      }}
      fluid
    >
      <Row>
        <Col xs={3} className="align-self-center">
          Date: {' '}
          <DateRangePicker
            small
            startDate={startDate} // momentPropTypes.momentObj or null,
            startDateId="" // PropTypes.string.isRequired,
            endDate={endDate} // momentPropTypes.momentObj or null,
            endDateId="" // PropTypes.string.isRequired,
            onDatesChange={({ startDate, endDate }) => updateDates(startDate, endDate)} // PropTypes.func.isRequired,
            focusedInput={focusedInput} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
            onFocusChange={focusedInput => setFocusedInput(focusedInput)} // PropTypes.func.isRequired,
          />
        </Col>
        <Col xs="fit-content" className="align-self-center">
          Price: {' '}
        </Col>
        <Col xs={2} className="align-self-center">
          <Range
            min={0}
            max={19}
            marks={{
              0: "0",
            }}
          />
        </Col>
      </Row>
    </Container>
  )
}

export default FilterBar;
