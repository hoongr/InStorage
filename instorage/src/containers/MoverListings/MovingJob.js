import React from 'react';
import {
  Container,
  Row,
  Col,
  Button,
} from 'react-bootstrap';

const MovingJob = (props) => {
  const {
    index,
    movingJob,
    onClick,
    onViewClick,
    changeLocation
  } = props;

  const {
    _moveDate,
    _moveTime,
    _inventoryPhotoURL,
    _hostAddress,
    _clientAddress,
    _distance
  } = movingJob;

  function  formatDate(isoDate) {
    const date = new Date(isoDate);

    return `${date.getMonth()+1}/${date.getDate()+1}`
  }

  function onLinkClick(e, index) {
    e.stopPropagation();
    onViewClick(index);
  }

  return (
    <Container
      style={{
        cursor: "pointer",
        height: 'auto',
        padding: '50px',
        borderBottom: "1px solid black",
      }}
      onClick={() => onClick(index)}
      onMouseOver={() =>{changeLocation(movingJob)}}
    >
      <style type="text/css">
        {`
          .btn-link:hover {
            color: #007bff !important;
          }
        `}
      </style>
      <Row className="justify-content-between">
        <Col className="align-self-center">
          <Row style={{ fontSize: '24px' }}>
            {_clientAddress}
          </Row>
          <Row style={{ fontSize: '20px' }}>
            <Button
              variant="link"
              onClick={(e) => onLinkClick(e, index)}
              style={{ padding: "0px", fontSize: "20px" }}
            >
              View Items
            </Button>
          </Row>
        </Col>
        <Col className="align-self-center">
          <Row style={{ fontSize: '28px' }} className="justify-content-end">
            {formatDate(_moveDate)}
          </Row>
          <Row style={{ fontSize: '20px' }} className="justify-content-end">
            {_moveTime}
          </Row>
          <Row style={{ fontSize: '20px' }} className="justify-content-end">
            {_distance} mi
          </Row>
        </Col>
      </Row>
    </Container>
  );
}

export default MovingJob;
