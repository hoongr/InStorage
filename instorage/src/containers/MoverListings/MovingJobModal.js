import React, { useState } from 'react';
import {
  Container,
  Row,
  Col,
  Button,
  Alert,
  Image,
} from 'react-bootstrap';
import { withFirebase } from '../../Firebase';
import FirebaseManager from '../../api/routes/firebaseManager';

const MovingJobModal = (props) => {
  const {
    job,
    firebase,
    closeModal,
    showToast,
  } = props;

  const {
    _movingJobID,
    _moverID,
    _clientAddress,
    _hostAddress,
    _moveDate,
    _moveTime,
    _inventoryPhotoURL,
  } = job;

  const {
    auth
  } = firebase;

  const [showAlert, setShow] = useState(false);

  function formatAddress(address) {
    return address.split(",")[0]
  }

  function formatDate(isoDate) {
    const date = new Date(isoDate);

    return `${date.getMonth()+1}/${date.getDate()+1}`
  }

  function onHelpClick() {
    const moverId = auth.currentUser.uid;
    const res = FirebaseManager.acceptMovingJob(_movingJobID, moverId);
    res.then(() => {
      setShow(true);
    })
  }

  return (
    <Container>
      <style type="text/css">
        {`
        .btn-a {
          background-color: #7CC6FE;
          color: white;
        }

        `}
      </style>
      <Row style={{ marginBottom: '25px' }}>
        <Col xs={8} className="align-self-center">
          <Row style={{ fontSize: '24px' }}>
            {formatAddress(_clientAddress)} to {formatAddress(_hostAddress)}
          </Row>
          <Row style={{ fontSize: '18px' }}>
            {formatDate(_moveDate)} | {_moveTime}
          </Row>
        </Col>
        <Col xs={4} className="align-self-center">
          <Row className="justify-content-end">
            <Button
              variant="a"
              onClick={onHelpClick}
            >
              Help Out!
            </Button>
          </Row>
        </Col>
      </Row>
      {(showAlert) && (
        <Alert variant="success" onClose={() => setShow(false)}>
          <Alert.Heading>Thank you for your help!</Alert.Heading>
          You'll receive an email reminding you about the job when the date is approaching!
        </Alert>
      )
      }
    </Container>
  )
}

export default withFirebase(MovingJobModal);
