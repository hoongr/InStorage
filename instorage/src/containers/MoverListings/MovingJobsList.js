import React, { useState, useEffect} from 'react';
import {
  Container,
  Row,
  Col,
  Modal,
  Image,
} from 'react-bootstrap';
import MovingJob from './MovingJob';
import MovingJobModal from './MovingJobModal';
import MapContainer from "../Map/MapContainer";
import FirebaseManager from "../../api/routes/firebaseManager";

const MovingJobsList = () => {
  const [movingJobs, setMovingJobs] = useState([]);
  const [showModal, setShowModal] = useState(false);
  const [showImageModal, setShowImageModal] = useState(false);
  const [currentListing, setCurrentListing] = useState(null);
  const [currentImage, setCurrentImage] = useState(null);
  const [lat, setLat] = useState(34.0689254);
  const [lng, setLng] = useState(-118.4473698)

  useEffect(() => {
    if (navigator && navigator.geolocation) {
      navigator.geolocation.getCurrentPosition(pos => {
        const res = FirebaseManager.getAllMovingJobs(pos.coords);

        res.then(movingJobs => setMovingJobs(movingJobs))
      });
    }
  }, []);

  function onJobClick(index) {
    setCurrentListing(movingJobs[index]);
    setShowModal(true);
  }

  function closeModal() {
    setShowModal(false);
  }

  function onViewClick(index) {
    setCurrentImage(movingJobs[index]._inventoryPhotoURL);
    setShowImageModal(true);
  }

  function closeImageModal() {
    setShowImageModal(false);
  }

  return (
    <Container
      fluid
    >
      <Modal
        centered
        size="lg"
        show={showModal}
        onHide={closeModal}
      >
        <Modal.Header closeButton>
        </Modal.Header>
        <Modal.Body>
          <MovingJobModal
            job={currentListing}
            closeModal={closeModal}
          />
        </Modal.Body>
      </Modal>
      <Modal
        centered
        size="lg"
        show={showImageModal}
        onHide={closeImageModal}
      >
        <Modal.Body>
          <Row className="justify-content-center">
            <Image fluid src={currentImage} />
          </Row>
        </Modal.Body>
      </Modal>
      <Row style={{ height: '100%' }}>
        <Col
          style={{
            height: '100%',
          }}
        >
          {(movingJobs.length > 0)
            ? (
              movingJobs.map(
                (movingJob, index) =>{
                  console.log(movingJob._clientAddress + " " + movingJob._distance);
                  return(<MovingJob
                    key={index}
                    index={index}
                    movingJob={movingJob}
                    onClick={onJobClick}
                    onViewClick={onViewClick}
                    changeLocation={(movingJob) => {
                      setLat(movingJob._coordinates[0]);
                      setLng(movingJob._coordinates[1]);
                    }}
                />)
              }
            ))
            : (
              <h5 className="pt-3">
                <div>
                  There are currently no moving jobs available.
                </div>
                <div className="pt-1">
                  Please check again later.
                </div>
              </h5>
            )
          }
        </Col>
        <Col>
          <Container
            fluid
            style={{
              position: 'fixed',
              width: '49%',
              height: '85%',
            }}
          >
              <MapContainer
                  currentLocation={{
                    lat: lat, 
                    lng: lng
                  }}
              ></MapContainer>
          </Container>
        </Col>
      </Row>


    </Container>
  );
}

export default MovingJobsList;
