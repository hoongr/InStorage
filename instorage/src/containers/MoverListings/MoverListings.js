import React, { useState, useEffect } from 'react';
import {
  Container,
  Row,
  Col,
} from 'react-bootstrap';

// import FilterBar from './FilterBar';
import MovingJobsList from './MovingJobsList';
import FirebaseManager from "../../api/routes/firebaseManager";

const MoverListings = () => {
  return (
    <Container
      style={{ padding: "0px" }}
      fluid
    >
      <MovingJobsList />
    </Container>
  )
}

export default MoverListings;
