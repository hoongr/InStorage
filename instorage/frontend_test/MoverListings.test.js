import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import FilterBar from "../src/containers/MoverListings/FilterBar"
import MoverListings from "../src/containers/MoverListings/MoverListings"
import MovingJob from "../src/containers/MoverListings/MovingJob"
import MovingJobModal from "../src/containers/MoverListings/MovingJobModal"
import MovingJobsList from "../src/containers/MoverListings/MovingJobsList"

describe('FilterBar', () => {
	const wrapper = shallow(<FilterBar />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})

describe('MoverListings', () => {
	const wrapper = shallow(<MoverListings />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})

describe('MovingJob', () => {
	const data = {
		_moveDate: "12/1/2019",
		_moveTime: "12:00PM",
		_hostAddress: "940 Tiverton Ave",
		_clientAddress: "532 Midvale Ave",
		_inventoryPhotoURL: "",
	};

	const wrapper = shallow(<MovingJob movingJob={data}/>);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})

describe('MovingJobModal', () => {
	const wrapper = shallow(<MovingJobModal />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})

describe('MovingJobsList', () => {
	const wrapper = shallow(<MovingJobsList />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})