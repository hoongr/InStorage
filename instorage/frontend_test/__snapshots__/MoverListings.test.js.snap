// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`MoverListings should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`MovingJob should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`MovingJobModal should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`MovingJobsList should match the snapshot 1`] = `ShallowWrapper {}`;
